File: src/main/java/framework/ftc/cobaltforge/Device.java
package framework.ftc.cobaltforge;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Injects the HardwareDevice
 * Created by Dummyc0m on 9/22/16.
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface Device {
    String value() default "";
}

File: src/main/java/framework/ftc/cobaltforge/kobaltforge/annotation/Device.kt
package framework.ftc.cobaltforge.kobaltforge.annotation


/**
 * Injects the HardwareDevice (only for directives)
 * Created by Dummyc0m on 9/22/16.
 */
@Retention(AnnotationRetention.RUNTIME)
annotation class Device(val value: String = "")
